### ISSUE TITLE
Add Support for Geyser/Floodgate

### ISSUE THREAD

**Issue by Tydium** [Nan1t/NanoLimbo#3]
Geyser / Floodgate is basically allowing Bedrock Players to join. With all my testing with this. For some reason there are a bunch of floating buttons, and the boss bar does not show up. Only the join message works. I have joined the discord server if you wan't more feedback.

**Comment by Tydium** [comment]
This may be more of the Geyser/Floodgate side of things so im going to also ask their devs to see if they can fix things regarding that

**Comment by Tydium** [comment]
I think I should redo this issue as a total. After some testing. The reason why it breaks is because the way floodgate working. Java edition works fine yet since floodgate doesn't exist on the fallback it doesn't work. Only way would probably be to use floodgate keys on server end.

**Comment by pwnker** [comment]
Geyser clients cannot connect at all when using LEGACY info forwarding on the velocity proxy.

**Comment by Tydium** [comment]
I managed to get this to somewhat work, however it isnt the best method at all. I turned forwarding off as a total then just made it seem like an offline mode server.

**Comment by pwnker** [comment]
Yeah setting forwarding to NONE and running the server in offline mode seems to allow both java and bedrock clients to connect without issue.

**Comment by Tydium** [comment]
I think that is a good solution for now. However make sure your backend is protected otherwise some idiot might try to join it.

**Comment by Nan1t** [comment]
When i have time, i will add normal support for Geyser. Sorry for no responses so long.

**Comment by Tydium** [comment]
All good @Nan1t . Take as much time as you need. Thanks for the great limbo server!

**Comment by MastoryMd5** [comment]
Sorry for the necro, with modern forwarding method and with floodgate in the proxy you should not have any issues.

**Comment by Tydium** [comment]
I actually stopped using this, but now that you mention it ill try it out thanks

**Comment by clarkkentz** [comment]
Geyers, Bedrock players can't join with Legacy forwarding mode.

**Comment by BrainRTP** [comment]
Thanks for the suggestion.

NanoLimbo targets vanilla Java Edition clients exclusively — Bedrock support would require translating every packet between Bedrock and Java protocols, and that's a substantial project on its own. Adding it here would contradict the "minimal, lightweight" goal of NanoLimbo (currently ~5 MB, near-zero allocations).

**What to do instead:**

Put [GeyserMC](https://geysermc.org/) *in front of* your proxy, not inside NanoLimbo. Two common setups:

1. **Velocity + Geyser-Velocity plugin**: Geyser runs as a Velocity plugin and translates Bedrock ↔ Java transparently. Velocity then routes the translated Java connection to NanoLimbo like any other client. This is the recommended path.
2. **Standalone Geyser**: run `Geyser-Standalone` on a separate port that acts as a Bedrock-speaking proxy in front of your Java proxy.

For cracked-Bedrock-as-Java authentication, add [Floodgate](https://geysermc.org/wiki/floodgate/) alongside Geyser.

Closing as out of scope. Feel free to reopen if you believe I've misunderstood the use case.
